executor/batch_point_get_test.go

```diff
@@ -162,3 +166,80 @@ func (s *testBatchPointGetSuite) TestBatchPointGetUnsignedHandleWithSort(c *C) {
 	tk.MustQuery("select id from t2 where id in (8738875760185212610, 1, 9814441339970117597) order by id").Check(testkit.Rows("1", "8738875760185212610", "9814441339970117597"))
 	tk.MustQuery("select id from t2 where id in (8738875760185212610, 1, 9814441339970117597) order by id desc").Check(testkit.Rows("9814441339970117597", "8738875760185212610", "1"))
 }
+
+func (s *testBatchPointGetSuite) TestBatchPointGetLockExistKey(c *C) {
+	var wg sync.WaitGroup
+	testLock := func(tk1, tk2 *testkit.TestKit, rc bool, key string, tableName string) {
+		doneCh := make(chan struct{}, 1)
+
+		tk1.MustExec("use test")
+		tk2.MustExec("use test")
+		tk1.MustExec("set session tidb_enable_clustered_index = 0")
+
+		tk1.MustExec(fmt.Sprintf("drop table if exists %s", tableName))
+		tk1.MustExec(fmt.Sprintf("create table %s(id int, v int, k int, %s key0(id, v))", tableName, key))
+		tk1.MustExec(fmt.Sprintf("insert into %s values(1, 1, 1), (2, 2, 2)", tableName))
+
+		if rc {
+			tk1.MustExec("set tx_isolation = 'READ-COMMITTED'")
+			tk2.MustExec("set tx_isolation = 'READ-COMMITTED'")
+		}
+
+		tk1.MustExec("begin pessimistic")
+		tk2.MustExec("begin pessimistic")
+
+		if !rc {
+			// lock exist key only for repeatable read
+			tk1.MustExec(fmt.Sprintf("select * from %s where (id, v) in ((1, 1), (2, 2)) for update", tableName))
+		} else {
+			// read committed will not lock non-exist key
+			tk1.MustExec(fmt.Sprintf("select * from %s where (id, v) in ((1, 1), (2, 2), (3, 3)) for update", tableName))
```

Comment:
We could add some `update` and `delete` cases here using `batchPointGet` as datasource to verify the lock behaviours.